src/App.tsx

```diff
@@ -1,24 +1,56 @@
 import React from 'react';
+import { Clock } from './components/Clock/Clock';
+
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: Math.round(Math.random() * 1000),
+  };
+
+  showClock = () => {
+    this.setState({
+      isClockVisible: true,
+    });
+  };
+
+  hideClock = () => {
+    this.setState({
+      isClockVisible: false,
+    });
+  };
+
+  renameClock = () => {
+    this.setState({
+      clockName: Math.round(Math.random() * 1000),
+    });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        <p>
+          Current time:
+          {' '}
+          {this.state.isClockVisible && <Clock clockName={this.state.clockName} />}
+        </p>
+
+        <button type="button" onClick={this.showClock}>
+          Show Clock
+        </button>
+
+        <button type="button" onClick={this.hideClock}>
+          Hide Clock
+        </button>
+
+        <button type="button" onClick={this.renameClock}>
+          Set random name
+        </button>
+      </div>
```

Comment:
If you want to improve your solution, you may change the disabled state of buttons if the clock is hidden/shown